components/clientComponents/forms/AddressComplete/AddressComplete.tsx

```diff
@@ -0,0 +1,309 @@
+"use client";
+import { AddressCompleteChoice, AddressCompleteProps, AddressElements } from "./types";
+import { getAddressCompleteChoices, getSelectedAddress } from "./utils";
+import { Combobox, Description, Label, ManagedCombobox } from "@clientComponents/forms";
+import { useState, useEffect, useRef } from "react";
+import { useTranslation } from "@i18n/client";
+import { useField } from "formik";
+import { cn } from "@lib/utils";
+import { SpinnerIcon } from "@serverComponents/icons/SpinnerIcon";
+import { Language } from "@lib/types/form-builder-types";
+import { countries } from "@lib/managedData/countries";
+
+interface ManagedComboboxRef {
+  changeInputValue: (value: string) => void;
+}
+
+export const AddressComplete = (props: AddressCompleteProps): React.ReactElement => {
+  const { id, name, required, ariaDescribedBy } = props;
+
+  const [field, meta, helpers] = useField(props);
+
+  const { t, i18n } = useTranslation("form-builder");
+
+  //Address Complete elements
+  const [choices, setChoices] = useState<string[]>([]);
+  const [addressResultCache, setAddressResultCache] = useState<AddressCompleteChoice[]>([]); // Cache the results from the address search.
+
+  // Cache and Allow values
+  const [isReady, setIsReady] = useState(false);
+  const [allow, setAllow] = useState(false);
+  const [apiKey, setApiKey] = useState("");
+
+  const toFullAddress = (address: AddressCompleteChoice): string => {
+    return address.Text + ", " + address.Description;
+  };
+
+  const comboboxRef = useRef<ManagedComboboxRef>(null);
+
+  // Check if addressComplete is allowed.
+  useEffect(() => {
+    const checkAllowed = async () => {
+      const response = await fetch("/api/components", { method: "POST" });
+      const jsonData = await response.json();
+      setAllow(jsonData.allowed);
+      setApiKey(jsonData.key);
+      setIsReady(true);
+    };
+
+    if (apiKey === "") {
+      // Don't recheck.
+      checkAllowed();
+    }
+  }, [apiKey]);
+
+  //Form fillers address elements
+  const [addressObject, setAddressObject] = useState<AddressElements | null>(null);
+
+  // Update the address object when the field value changes
+  useEffect(() => {
+    if (field.value) {
+      try {
+        const parsedValue = JSON.parse(field.value);
+        // Only update if parsedValue is different from addressObject
+        if (JSON.stringify(parsedValue) !== JSON.stringify(addressObject)) {
+          setAddressObject(parsedValue);
+        }
+      } catch (e) {
+        if (addressObject !== null) {
+          setAddressObject(null);
+        }
+      }
+    }
+  }, [field.value]);
+
+  // Update the field value when the address object changes
+  useEffect(() => {
+    const newValue = addressObject ? JSON.stringify(addressObject) : "";
+    // Only update if newValue is different from field.value
+    if (newValue !== field.value) {
+      helpers.setValue(newValue);
+    }
+  }, [addressObject, helpers]);
```

Comment:
[Continuing from above] You can see the problem with this code if you update your useEffect dependencies to clear the typescript warning, the bug you were trying to fix comes back.

By initializing the object as described in the previous comment, you can remove the newValue/oldValue check from this second useEffect as well.